website/docs/migration/from_state_notifier.mdx

```diff
@@ -0,0 +1,264 @@
+This happens because `(Async)Notifier`s lacks a `mounted` property, which was available on
+`StateNotifier`.  
+Considering their difference in lifecycle, this makes perfect sense; while possible, a `mounted`
+property would be misleading on the new notifiers: `mounted` would *almost always* be `true`.
+
+While it would be possible to craft a [custom workaround](https://github.com/rrousselGit/riverpod/issues/1879#issuecomment-1303189191),
+it's reccomended to work around this by canceling the asynchronous operation.
+
+Canceling an operation can be done with a custom [Completer](https://api.flutter.dev/flutter/dart-async/Completer-class.html),
+or any custom derivative.
+
+For example, if you're using `Dio` to perform network requests, consider using a [cancel token](https://pub.dev/documentation/dio/latest/dio/CancelToken-class.html).
+As explained [here](/docs/concepts/modifiers/auto_dispose#example-canceling-http-requests-when-no-longer-used),
+a `CancelToken` can be used to cancel requests.
+
+Therefore, the above example migrates to the following:
+<AutoSnippet language="dart" {...oldLifecyclesFinal}></AutoSnippet>
+
+
+## Mutations APIs are the same as before
+
+Up until now we've shown the differences between `StateNotifier` and the new APIs.  
+Instad, one thing `Notifier`, `AsyncNotifer` and `StateNotifier` share is how their states
+can be consumed and mutated.
+
+Consumers can obtain data from these three providers with the same syntax, which is great in case
+you're migrating away from `StateNotifier`, and this applies for notifiers methods, too.
+
+<AutoSnippet language="dart" {...consumersDontChange}></AutoSnippet>
+
+## Other migrations
+
+Let's explore the less-impactful differences between `StateNotifier` and `Notifier` (or `AsyncNotifier`)
+
+### From `.addListener` and `.stream`
+
+`StateNotifier`'s `.addListener` and `.stream` can be used to listen for state changes.
+These two APIs are now to be considered outdated.
+
+This is intentional due to the desire to reach full API uniformity with `Notifier`, `AsyncNotifier` and other providers.  
+Indeed, using a `Notifier` or an `AsyncNotifier` shouldn't be any different from any other provider.
+
+Therefore this:
+<AutoSnippet language="dart" {...addListenerOld}></AutoSnippet>
+
+Becomes this:
+<AutoSnippet language="dart" {...addListener}></AutoSnippet>
+
+In a nutshell: if you want to listen to a `Notifier`/`AsyncNotifer`, just use `ref.listen`,
+[as explained here](/docs/concepts/reading#using-reflisten-to-react-to-a-provider-change).
+
+### From `.debugState` in tests
+
+`StateNotifier` exposes `.debugState`: this property is used for pkg:state_notifier users to enable
+state access from outside the class when in development mode, for testing purposes.
+
+If you're using `.debugState` to access state in tests, chances are that you need to drop this
+approach.
+
+`Notifier` / `AsyncNotifer` don't have a `.debugState`; instead, they expose `.state`, which is
```

Comment:
Could be worth having a code snippet here, with a showcase on how to get the notifier